hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/DecodingFacade.java

```diff
@@ -1,33 +1,255 @@
 package com.hedera.services.store.contracts.precompile;
 
+import com.esaulpaugh.headlong.abi.Function;
+import com.esaulpaugh.headlong.abi.Tuple;
 import com.google.protobuf.ByteString;
+import com.hedera.services.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.AccountAmount;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.CryptoTransferTransactionBody;
+import com.hederahashgraph.api.proto.java.NftTransfer;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TokenTransferList;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.datatypes.Address;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
-import static com.hedera.services.utils.EntityIdUtils.tokenParsedFromSolidityAddress;
-
 @Singleton
 public class DecodingFacade {
 	private static final int ADDRESS_BYTES_LENGTH = 20;
 	private static final int ADDRESS_SKIP_BYTES_LENGTH = 12;
 	private static final int FUNCTION_SELECTOR_BYTES_LENGTH = 4;
 
+	private static final Function CRYPTO_TRANSFER_FUNCTION = new Function("cryptoTransfer((bytes32,(bytes32,int64)[],(bytes32,bytes32,int64)[])[])",
+			"((bytes32,(bytes32,int64)[],(bytes32,bytes32,int64)[])[])");
+	private static final Function TRANSFER_TOKENS_FUNCTION = new Function("transferTokens(address,address[],int64[])",
+			"(bytes32," +
+					"bytes32[],int64[])");
```

Comment:
```suggestion
					"bytes32[],int64[])");
```
why not "(bytes32,bytes32[],int64[])" instead of creating two Strings and concatenating them?